src/MongoDB.Driver.Core/Core/Authentication/AwsSignatureVersion4.cs

```diff
@@ -0,0 +1,203 @@
+    {
+        /// <summary>
+        /// Gets canonical headers.
+        /// </summary>
+        /// <param name="requestHeaders">The request headers.</param>
+        /// <returns>The canonical headers.</returns>
+        public static string GetCanonicalHeaders(SortedDictionary<string, string> requestHeaders)
+        {
+            return string.Join("\n", requestHeaders.Select(x => $"{x.Key.ToLowerInvariant()}:{x.Value}"));
+        }
+
+        /// <summary>
+        /// Gets Amazon region from host.
+        /// </summary>
+        /// <param name="host">The host.</param>
+        /// <returns>The region.</returns>
+        public static string GetRegion(string host)
+        {
+            if (host == "sts.amazonaws.com")
+            {
+                return "us-east-1";
+            }
+
+            var split = host.Split('.');
+            if (split.Count() > 1)
+            {
+                return split[1];
+            }
+
+            return "us-east-1";
+        }
+
+        /// <summary>
+        /// Gets signed headers.
+        /// </summary>
+        /// <param name="requestHeaders">The request headers.</param>
+        /// <returns>The signed headers.</returns>
+        public static string GetSignedHeaders(SortedDictionary<string, string> requestHeaders)
+        {
+            return string.Join(";", requestHeaders.Keys.Select(x => x.ToLowerInvariant()));
+        }
+
+        /// <summary>
+        /// Sign request.
+        /// </summary>
+        /// <param name="accessKey">The access key.</param>
+        /// <param name="secretKey">The secret key.</param>
+        /// <param name="securityToken">The security token.</param>
+        /// <param name="salt">The salt.</param>
+        /// <param name="host">The host.</param>
+        /// <returns>The signed request.</returns>
+        public static Tuple<string, string> SignRequest(
+            string accessKey,
+            string secretKey,
+            string securityToken,
+            byte[] salt,
+            string host)
+        {
+            return SignRequest(DateTime.UtcNow, accessKey, secretKey, securityToken, salt, host);
+        }
+
+        /// <summary>
+        /// Signs the request.
+        /// </summary>
+        /// <param name="dateTime">The date time.</param>
+        /// <param name="accessKey">The access key.</param>
+        /// <param name="secretKey">The secret key.</param>
+        /// <param name="securityToken">The security token.</param>
+        /// <param name="salt">The salt.</param>
+        /// <param name="host">The host.</param>
+        /// <returns>The signed request.</returns>
+        public static Tuple<string, string> SignRequest(
+            DateTime dateTime,
+            string accessKey,
+            string secretKey,
+            string securityToken,
+            byte[] salt,
+            string host)
+        {
+            var body = "Action=GetCallerIdentity&Version=2011-06-15";
```

Comment:
can you clarify this line?

Reply:
I saw this line in the doc. This is weird, but at the same time this is what the doc requires